src/storage/storage.rs

```diff
@@ -154,9 +157,12 @@ where
     S: Storage,
     H: HashFunctionType,
 {
-    #[allow(unused)]
     pub fn new(storage: S) -> Self {
-        Self { storage, _h: Default::default() }
+        Self { storage: Arc::new(Mutex::new(storage)), _h: Default::default() }
+    }
+
+    pub async fn storage(&self) -> tokio::sync::MutexGuard<S> {
```

Comment:
I'd consider renaming this to convey/warn that it's locking, something like `acquire_storage()` or similar